ducktape/decorate.py

```diff
@@ -0,0 +1,229 @@
+# Copyright 2015 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import functools
+
+PARAMETRIZED = "PARAMETRIZED"
+INJECTED = "INJECTED"
+MATRIX = "MATRIX"
+MARKS = {PARAMETRIZED, INJECTED, MATRIX}
+
+_marked_functions = {}
```

Comment:
You could also use the `__dict__` field on the unbound methods to store the information about what's been marked, which would avoid this global data.